### ISSUE TITLE
Custom Object Mapper configuration for Data Wrapper breaks swagger-ui.

### ISSUE THREAD

**Issue by langley-agm** [springfox/springfox#1392]
Hi,

I'm using 2.5.0.

I'm not sure if this qualifies as a Question or an Issue.

I have a custom Object Mapper configuration that breaks the swagger ui.

I'm wrapping all my rest responses inside a wrapper data object by modifying a property of the Object Mapper: 

	@Bean
	public ObjectMapper dataWrapperObjectMapper(Jackson2ObjectMapperBuilder builder) {
		ObjectMapper mapper = builder.build();
		return mapper.setConfig( mapper.getSerializationConfig().withRootName("data") );
	}

This is affecting the endpoints created by springfox and breaking the schema expected by the swagger ui, hence showing just an error. For example, `/swagger-resources/configuration/security` is showing:

	{
	  "data" : {
		"apiKeyVehicle" : "header",
		"scopeSeparator" : ",",
		"apiKeyName" : "api_key"
	  }
	}

![image](https://cloud.githubusercontent.com/assets/10603341/16902879/81bd81d8-4c32-11e6-8000-cb466237fc78.png)

of course the `/v2/api-docs` are wrapped as well.

I tried using an `ApplicationListener<ObjectMapperConfigured>` and I get the same results. Any guidance would be greatly appreaciated.

**Comment by dilipkrish** [comment]
This is a known issue. Also reproduced via similar configurations as #597

**Comment by langley-agm** [comment]
Are there any ideas yet for correctly fixing this? If these endpoints are created only for springfox consume, I wonder if SpringFox should create and use its own mapper or serializer and not use the one customized by the users which can have infinite variables, maybe its own dispatcher as well.

Another thing I can think of is to check for the mapper's root name and userRootName setting, and if it does, use this setting to unwrap the contents of the json files springfox creates when using them to configure the ui and stuff.

**Comment by dilipkrish** [comment]
You're right, it is a very esoteric setup in-itself which is why not much has been done towards fixing it. By esoteric I mean
- How many use the bundled swagger-ui?
- How many of those that use the bundled swagger-ui configure the object mappers to have a wrapping root node?

That is not to say it shouldn't be isolated and fixed. Would be great to get a fix for this, and Im not sure what the right solution is yet. My gut instinct is to create a vnd media type to isolate the [configuration endpoints](https://github.com/springfox/springfox/blob/master/springfox-swagger-common/src/main/java/springfox/documentation/swagger/web/ApiResourceController.java#L35) and setup a configuration to register the http message converter to handle it.

**Comment by langley-agm** [comment]
"How many of those that use the bundled swagger-ui configure the object mappers to have a wrapping root node".

I don't think its about this specific configuration, its about a custom configuration in the client's app leaking into springfox implementation, not swagger-ui, springfox-swagger and springfox-swagger-ui. This specific one is affecting it really bad, but there could be many others that can affect it in a minor or major degree. I don't think these should be leaked at all independently of the degree in which they affect.

**Comment by langley-agm** [comment]
For reference, I wonder if another option could be adding custom serializers for the specific springfox classes like `SecurityConfiguration`, `UiConfiguration`, `SwaggerResource`, etc. and render them the way the library itself will expect them to be rendered as. Either that or add the ability to easily add a custom deserializer in the springfox-swagger-ui side.

**Comment by dilipkrish** [comment]
I like the idea of custom serializers. May be an easier fix than creating a separate media type.

**Comment by langley-agm** [comment]
We can scratch the custom serializer option, I tried a custom serializer but Spring will still add the wrapper to it.

**Comment by Poeschl** [comment]
Any process on that bug?

**Comment by dilipkrish** [comment]
@Poeschl ☝️ looking for contributions.
